esmvaltool/diag_scripts/kcs/resampling.py

```diff
@@ -7,40 +7,45 @@
 This provides eight resampled EC-Earth time series for each of the scenarios.
 """
 
-from itertools import combinations
+from itertools import product
 
 import pandas as pd
 import xarray as xr
+import numpy as np
 
-from esmvaltool.diag_scripts.shared import (get_diagnostic_filename,
-                                            get_plot_filename, run_diagnostic,
+from esmvaltool.diag_scripts.shared import (run_diagnostic,
                                             select_metadata)
 
 
-def segment(dataset, start_year, end_year, block_size):
+def segment(dataset, period, step=5):
     """Segment (part of) a dataset into x blocks of n years each.
 
     Returns a new dataset with additional coordinate 'segment'.
     """
-    segmented_dataset = xr.concat([
-        ds.sel(time=slice(str(year), str(year+block_size)))
-        for year in range(start_year, end_year, block_size)
-    ], dim='segment')
+
+    for year in range(*period, step):
+        segmented_dataset = xr.concat(
+            [dataset.sel(time=slice(str(year), str(year+step)))], dim='segment'
+        )
```

Comment:
@SarahAlidoost Be advised that this breaks the code. See this example of [list comprehension](https://www.programiz.com/python-programming/list-comprehension). If  you want to replace the implicit loop by an explicit loop, you would need to add all segments to a list first, and then do the `xr.concat`:

```suggestion
    segments = []
    for year in range(*period, step):
        segment = dataset.sel(time=slice(str(year), str(year+step)))
        segments.append(segment)
    segmented_dataset = xr.concat(segments, dim='segment' )
```
Currently, you're overwriting `segmented_dataset` in each iteration, which is not what you want.